### DISCUSSION TITLE
showUserLocation is not working despite putting in the run time location permission

### DISCUSSION THREAD

**Discussion by elkee2003** [react-native-maps/react-native-maps#4720] [Q&A] [answered]
### Summary

User location isn't showing despite the fact I followed the instructions- I believe
![Screenshot  focused(17)](https://user-images.githubusercontent.com/42178058/235154344-c8f760ee-9a6d-4a2b-8ec0-8980209868a0.jpg)

### Reproducible sample code

```TSX
import { View, Text } from 'react-native'
import React from 'react'
import MapView, {PROVIDER_GOOGLE} from 'react-native-maps'
import MapViewDirections from 'react-native-maps-directions';

const origin = {latitude: 37.3318456, longitude: -122.0296002};
const destination = {latitude: 37.771707, longitude: -122.4053769};
const GOOGLE_MAPS_APIKEY = 'xxxxxxxxxxaF_F1ZjW_ZS7GD_pj';

const HomeScreen = () => {
  return (
    <View>
      <MapView
        style={{height:'100%', width:'100%'}}
        provider={PROVIDER_GOOGLE}
        showsUserLocation={true}
        initialRegion={{
          latitude: 28.450627,
          longitude: -16.263045,
          latitudeDelta: 0.015,
          longitudeDelta: 0.0121,
        }}
       >
        <MapViewDirections
            origin={origin}
            destination={destination}
            apikey={GOOGLE_MAPS_APIKEY}
        />
      </MapView>
    </View>
  )
}

export default HomeScreen;
```

### Steps to reproduce

I imported MapView
I deisplayed my map with longitude , latitude etc
I went to android, app, src, main, andriodmanifest.xml and added:
<uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
    <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
I came back to my file and added showuserlocation = {true} in my code where i imported mapview.

Despite all these, it didn't show.

### Expected result

For User location to be shown

### Actual result

it's not showing the userlocation on my map
You can see it's not showing the user location:
![Screenshot (16)](https://user-images.githubusercontent.com/42178058/235152162-6e19a487-1179-4dc3-8f97-db4b303c15d0.png)

Here I add the permissions:
![Screenshot  focused(17)](https://user-images.githubusercontent.com/42178058/235152635-9953b161-03e9-412c-9d6f-5aa3efc34716.jpg)

Here I accessed the showUserLocation to be true:
![Screenshot focused (18)](https://user-images.githubusercontent.com/42178058/235205426-0d6a3f6f-9b1d-47e5-a329-b6bd15c4ffbe.png)

### React Native Maps Version

1.7.1

### What platforms are you seeing the problem on?

Android

### React Native Version

0.73.9

### What version of Expo are you using?

Not using Expo

### Device(s)

Tecno Camon 19

**Comment by stjasink** [comment, marked as answer]
Try adding in a call to actually request the permission from the user before rendering the map, e.g.:

```
  request(PERMISSIONS.ANDROID.ACCESS_FINE_LOCATION).then(result => {
     console.log('Got permission: ' + result);
  });

```

**Reply by elkee2003** [reply]
Thank you
